src/main/java/htsjdk/samtools/read/Read.java

```diff
@@ -0,0 +1,1438 @@
+     *
+     * @return name of the contig this read is mapped to; {@link ReadConstants#NO_ALIGNMENT_REFERENCE_NAME} if unset (e.g. for unmapped reads).
+     */
+    @Override
+    String getContig();
+
+    /**
+     * Gets the 1-based inclusive leftmost position of the sequence remaining after clipping.
+     *
+     * <p>Equivalent to <b>POS</b> in the <a href="http://samtools.github.io/hts-specs/SAMv1.pdf">SAM specifications</a>.
+     *
+     * @return 1-based start position; {@link ReadConstants#NO_ALIGNMENT_START} if there is no position (e.g. for unmapped read).
+     */
+    @Override
+    int getStart();
+
+    /**
+     * Gets the 1-based inclusive rightmost position of the sequence remaining after clipping.
+     *
+     * @return 1-based closed-ended position; {@link ReadConstants#NO_ALIGNMENT_START} if there is no position (e.g. for unmapped read).
+     */
+    @Override
+    int getEnd();
+
+    /**
+     * Sets the alignment position.
+     *
+     * <p>Equivalent to <b>RNAME</b> and <b>POS</b> in the <a href="http://samtools.github.io/hts-specs/SAMv1.pdf">SAM specifications</a>.
+     *
+     * @param contig name of the contig this read is mapped to; {@link ReadConstants#NO_ALIGNMENT_REFERENCE_NAME} if unset (e.g. for unmapped reads).
+     * @param start 1-based start position; {@link ReadConstants#NO_ALIGNMENT_START} if there is no position (e.g. for unmapped read).
+     *
+     * @throws IllegalArgumentException if the contig is {@code null}.
+     */
+    void setAlignmentPosition(final String contig, final int start);
+
+    /**
+     * Sets the alignment position.
+     *
+     * <p>Equivalent to <b>POS</b> and <b>POS</b> in the <a href="http://samtools.github.io/hts-specs/SAMv1.pdf">SAM specifications</a>.
+     *
+     * <p>Default implementation uses the {@link #setAlignmentPosition(String, int)}, ignoring the {@link Locatable#getEnd()}.
+     *
+     * @param position location this read is mapped to.
+     *
+     * @throws IllegalArgumentException if the locatable is {@code null}.
+     */
+    default void setAlignmentPosition(final Locatable position) {
+        if (position == null) {
+            throw new IllegalArgumentException("null position");
+        }
+        setAlignmentPosition(position.getContig(), position.getStart());
+    }
+
+    /**
+     * Gets the read length.
+     *
+     * <p>Note: this is not necessarily the same as the number of reference bases the read is aligned to.
+     *
+     * @return The number of bases in the read
+     */
+    int getLength();
+
+    /**
+     * Checks if the read has bases.
+     *
+     * <p>Default implementation returns {@code getLength() == 0}.
+     *
+     * @return {@code true} if the read has no bases; {@code false} otherwise.
+     */
+    default boolean isEmpty() {
+        return getLength() == 0;
+    }
+
+    /**
+     * Gets the PHERD scaled mapping quality.
```

Comment:
PHERD->PHRED

Reply:
also, it's not capitalized normally: `phred`

Reply:
Link to MAPQ in the sam spec

Reply:
Opps.

Reply:
Done.